### ISSUE TITLE
error: no matching function for call to 'find_if'

### ISSUE THREAD

**Issue by dennisklein** [FairRootGroup/FairMQ#29]
With Boost 1.64 everything works, but with Boost 1.66 and 1.67:

```
[  0%] Building CXX object fairmq/CMakeFiles/FairMQ.dir/DeviceRunner.cxx.o
In file included from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/Tools.h:14:0,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/Plugin.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/PluginManager.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.h:13,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.cxx:9:
/home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/tools/Network.h: In function ‘std::__cxx11::string fair::mq::tools::getIpFromHostname(const string&)’:
/home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/tools/Network.h:143:10: error: no matching function for call to ‘find_if(boost::asio::ip::basic_resolver<boost::asio::ip::tcp>::results_type, boost::asio::ip::basic_resolver<boost::asio::ip::tcp>::iterator&, fair::mq::tools::getIpFromHostname(const string&)::<lambda(const endpoint&)>)’
         });
          ^
In file included from /usr/include/c++/7/algorithm:62:0,
                 from /home/dklein/alibuild/sw/fedora27_x86-64/boost/v1.66.0-1/include/boost/any.hpp:17,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/EventManager.h:22,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.cxx:9:
/usr/include/c++/7/bits/stl_algo.h:3923:5: note: candidate: template<class _IIter, class _Predicate> _IIter std::find_if(_IIter, _IIter, _Predicate)
     find_if(_InputIterator __first, _InputIterator __last,
     ^~~~~~~
/usr/include/c++/7/bits/stl_algo.h:3923:5: note:   template argument deduction/substitution failed:
In file included from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/Tools.h:14:0,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/Plugin.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/PluginManager.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.h:13,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.cxx:9:
/home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/tools/Network.h:143:10: note:   deduced conflicting types for parameter ‘_IIter’ (‘boost::asio::ip::basic_resolver_results<boost::asio::ip::tcp>’ and ‘boost::asio::ip::basic_resolver_iterator<boost::asio::ip::tcp>’)
         });
          ^
/home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/tools/Network.h: In function ‘std::__cxx11::string fair::mq::tools::getIpFromHostname(const string&, boost::asio::io_service&)’:
/home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/tools/Network.h:169:10: error: no matching function for call to ‘find_if(boost::asio::ip::basic_resolver<boost::asio::ip::tcp>::results_type, boost::asio::ip::basic_resolver<boost::asio::ip::tcp>::iterator&, fair::mq::tools::getIpFromHostname(const string&, boost::asio::io_service&)::<lambda(const endpoint&)>)’
         });
          ^
In file included from /usr/include/c++/7/algorithm:62:0,
                 from /home/dklein/alibuild/sw/fedora27_x86-64/boost/v1.66.0-1/include/boost/any.hpp:17,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/EventManager.h:22,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.cxx:9:
/usr/include/c++/7/bits/stl_algo.h:3923:5: note: candidate: template<class _IIter, class _Predicate> _IIter std::find_if(_IIter, _IIter, _Predicate)
     find_if(_InputIterator __first, _InputIterator __last,
     ^~~~~~~
/usr/include/c++/7/bits/stl_algo.h:3923:5: note:   template argument deduction/substitution failed:
In file included from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/Tools.h:14:0,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/Plugin.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/PluginManager.h:12,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.h:13,
                 from /home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/DeviceRunner.cxx:9:
/home/dklein/alibuild/sw/SOURCES/FairMQ/v1.2.0/v1.2.0/fairmq/tools/Network.h:169:10: note:   deduced conflicting types for parameter ‘_IIter’ (‘boost::asio::ip::basic_resolver_results<boost::asio::ip::tcp>’ and ‘boost::asio::ip::basic_resolver_iterator<boost::asio::ip::tcp>’)
         });
          ^
make[2]: *** [fairmq/CMakeFiles/FairMQ.dir/build.make:63: fairmq/CMakeFiles/FairMQ.dir/DeviceRunner.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1113: fairmq/CMakeFiles/FairMQ.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

**Comment by rbx** [comment]
This is already fixed in de7ddc0.

**Comment by dennisklein** [comment]
Tested 1.66 and 1.67 to confirm. Released in [1.2.1](https://github.com/FairRootGroup/FairMQ/releases/tag/v1.2.1).
